### ISSUE TITLE
Instability when training

### ISSUE THREAD

**Issue by jaylee2000** [AndreasBergmeister/ram#4]
Hi, thanks for open-sourcing your great work.
I'm trying to reproduce your results on GenEval fine-tuning.

However, the runs are unstable & fail to reach 0.97:
1. The first run reached 0.90x, but crashed catastrophically after I resumed from a saved checkpoint.
2. The second run reached 0.94x, but plateaued and then collapsed.

Both runs used the same code (*) and random seed.

<img width="1034" height="553" alt="Image" src="https://github.com/user-attachments/assets/7ffd4a6b-968b-478f-b1b2-2e96e00c0814" />

Is it expected to be sensitive to random seeds, non-determinism from multi-GPU training, etc.?
Or is it more likely that there's something wrong on my side?

*: I am running on a server with CUDA 13.0 installed, so I have
1. Updated PyTorch and CUDA package versions
2. Patched code under `reward_models/` to handle the key name convention mismatch for Mask2Former

The full `git diff` is visible [here](https://gist.github.com/jaylee2000/4be79d6683f1b6187f989b23a989f485); I did not make any changes to the core training code.

**Comment by lian700** [comment]
Will other rewards also encounter this issue?

**Comment by AndreasBergmeister** [comment]
Thanks for the report. This isn't expected. My own runs were pretty consistent.

One thing that might be relevant: before open-sourcing, I let Claude Code refactor the repo, which included updating package versions and streamlining the install. I should have checked more carefully that nothing changed functionally, and it's possible something slipped in during that pass.

Our cluster is down right now, so I can't reproduce immediately, but I'll look into it asap and report here. Sorry for the trouble.

**Comment by jaylee2000** [comment]
I figured out that that CUDA 13 packages were the culprit; this problem does not occur when running the code with a conda environment with CUDA 12.x toolkits installed (works for servers with CUDA 13.0 installed).

@AndreasBergmeister With the correct packages installed, RAM works surprisingly well! Impressed with the results :)
